### ISSUE TITLE
HDUList.pop doesn't work with strings

### ISSUE THREAD

**Issue by astrofrog** [astropy/astropy#7229]
HDUList behaves like a dictionary when it comes to getting HDUs:

```python
>>> hdulist['SEDS']
```

However, ``pop`` doesn't work with strings:

```python
>>> hdulist.pop('SEDS')
...
TypeError: 'str' object cannot be interpreted as an integer
```

It would be nice to add this since the meaning should be pretty unambiguous.

**Comment by pllim** [comment]
Tuple doesn't work either. Tuple is needed when there are multiple extensions with the same name; for example: `('SCI', 1)`

**Comment by saimn** [comment]
Implemented in #7236.
